### PR TITLE
Api call

### PR DESCRIPTION
**Pull request by WujiMikeChen**
added test files and made code more testable

### CODE REVIEW

File: src/search_recipe/QueryAPI.java [hunk 1]
```diff
@@ -15,10 +15,10 @@ public class QueryAPI {
     /*
     This calls the API with the passed in arraylist of ingredients and the number of results you want to return
      */
-    public static JsonArray getResults(ArrayList<String> ingredients, int number) throws Exception {
+    public static JsonArray getResults(ArrayList<String> ingredients, int number) {
```
Comment (JohnZHANGFZ) [inline src/search_recipe/QueryAPI.java:18]: Is it possible for the API to throw exceptions? If Yes then how we are going to deal with this case?

File: src/database_creation/CreateDatabase.java [hunk 2]
```diff
@@ -7,6 +7,10 @@
 import java.io.*;
 import java.util.ArrayList;
 
+/*
+This code solely exists to create the database and will never be run in the program
+ */
+
 public class CreateDatabase {
     final ArrayList<String> ingredients;
```
Comment (lexazhs) [inline src/database_creation/CreateDatabase.java:15]: make them private?

**Review by lexazhs — COMMENTED** [review]
Overall looks good to me, consider a small change

File: src/search_recipe/QueryAPI.java [hunk 3]
```diff
@@ -66,21 +66,21 @@ public static JsonArray ParseString(String response) {
...
         BufferedReader reader;
 
         try {
-            reader = new BufferedReader(new FileReader("APIKey.txt"));
+            reader = new BufferedReader(new FileReader(filename));
             APIKey = reader.readLine();
             reader.close();
         } catch (IOException e) {
-            System.out.println("Check your key file");
+            throw new IOException();
```
Comment (cher-ning) [inline src/search_recipe/QueryAPI.java:78]: consider changing to file not found exception instead

File: src/search_recipe/QueryAPI.java [hunk 4]
```diff
@@ -66,21 +66,21 @@ public static JsonArray ParseString(String response) {
...
     }
 
-    public static String getAPIKey() {
+    public static String getAPIKey(String filename) throws IOException {
         String APIKey = "";
         BufferedReader reader;
 
         try {
-            reader = new BufferedReader(new FileReader("APIKey.txt"));
+            reader = new BufferedReader(new FileReader(filename));
```
Comment (JohnZHANGFZ) [inline src/search_recipe/QueryAPI.java:79]: Good! read the APIKey in this way makes it more private.

**Review by JohnZHANGFZ — CHANGES_REQUESTED** [review]

**Review by lexazhs — APPROVED** [review]

**Pull request merged**